### DISCUSSION TITLE
【メンテナー必読: 共有事項】バージョン1リリースまでにすること

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#52] [【メンテナー必読】共有事項]
- Project: https://github.com/orgs/first-contributions-ja/projects/2
- Milestones:
  - [ロゴ・関連画像が準備後のタスク Milestone](https://github.com/first-contributions-ja/first-contributions-ja.github.io/milestone/2)
  - [ドッグフーディング後のタスク Milestone](https://github.com/first-contributions-ja/first-contributions-ja.github.io/milestone/3)

---

- バージョン1リリースまでの間に全員に共有すべき内容があればこちらに投稿

**Comment by pss-aileen** [comment]
# コーディングにまつわること

- 増えてきたらコーディング規約にまとめる

**Reply by pss-aileen** [reply]
## `<Image>` の `src` は `${BASE_PATH}/xxx.svg` とすること

```tsx
<Image src={`${BASE_PATH}/vercel.svg`}
```
> `< Image >`で画像を扱うときは、Github Pagesにデプロイする関係で、basePathをimportして、srcのパスの先頭につけないといけないことだけ、注意しておいてください！

#### 参考
- https://github.com/first-contributions-ja/first-contributions-ja.github.io/pull/37#discussion_r1524798865

**Reply by pss-aileen** [reply]
## Tailwind CSSは そのまま `className` に書く
- CSSをまとめて使い回したりを考えていたが、これらは最後にすることになった
- そのため `className` にガシガシ書き足す方針で

**Reply by kagomen** [reply]
## アイコンを使用する

Material-Iconsを導入しました。 #86 

### 使用方法
使用するページで`import`してから使用します。
`className`を指定して直接スタイル当てられます。
```
import TwitterIcon from '@mui/icons-material/Twitter';

 <TwitterIcon fontSize="large" className="text-white"/>
```
 
参考: https://mui.com/material-ui/material-icons/

**Reply by pss-aileen** [reply]
## ブランチ命名規則の変更
ブランチ名を `_` で区切っていましたが、`-` で区切るよう変更しました

```
docs/#01-create-documents
```
参考: https://github.com/first-contributions-ja/first-contributions-ja.github.io/pull/96#discussion_r1536722958

**Reply by kagomen** [reply]
## マークダウン記入時の規則の追加

- 英単語の前後は半角空白を入れる
- なるべく正式名称を使う

参考 https://github.com/first-contributions-ja/first-contributions-ja.github.io/issues/103#issuecomment-2022440760

**Comment by pss-aileen** [comment]
# デザインに関すること

**Reply by pss-aileen** [reply]
## 実装イメージ

2024/03/18 ちょっと更新しました

![スクリーンショット 2024-03-18 1 16 52](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/e75bbd10-30c3-4363-8946-930a0b3bcbf8)


<details>
<summary>過去の実装イメージ</summary>

![313291482-bad6cb8f-cae9-4dea-9691-ad89cec3c63d](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/f81d4ac8-45a0-43ce-853e-9cd1ba56fe0c)

</details>


- デザインについて
  - 「手順」の左の白丸はアイコン決めれなかったので丸にしているだけです。適切なアイコンをお願いいたします🙇‍♀
  - サイトタイトルの左側の●もとりあえず入れているだけなので、プロジェクト独自のロゴや何かができれば差し替える。とりあえず●ですすめてみる 
  -  イメージなので、好きに作り変えていただいてOKです！
  - 日本っぽくとりあえず赤色採用
  - とりあえずシンプルにしてます（後にわちゃわちゃにすれば良いかと）
- 素材: https://www.openpeeps.com/


#### 参考
- #46

**Reply by pss-aileen** [reply]
## ページ構成について

- TOPだけ作成
  - **ヒーローエリア**
    - CTAなしで
    - イラストと
    - 見出し、リード
  - **「コントリビューションの簡単7ステップ」コントリビュートの方法の概要**
    - **目次だけをまとめる！！**
    - README.mdに飛ばす（CTA）
    - 「詳しい手順はこちら」
  - **ヘッダー**
    - サイトタイトル（TOPのリンク付き、もしくリンクなしでも良いかも）のみで
    - 初期のカンプのヘッダーの右側にあるグローバルナビ（HOME、ABOUT）は実装しない方向になった
    - ざっくり初期カンプ
    - <img src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/b31eb0f3-b732-41ff-a872-fba96b4f9162" width="400">

---

- 個人個人のカードをどうするか...
  - とりあえずはどこにも表示させない！！！
  - contribution.json に書き加えるというのができればOK
  - シングルページ的ものか、一覧を表示させるかは要相談

---

- ABOUTは消す方向で

**Reply by pss-aileen** [reply]
## カラー
- メインカラー: #DC2626  TailwindCSS: Red600
- テキストカラー: #18181b TailwindCSS: Zinc900

**Reply by kagomen** [reply]
すみません、こちらテキストのグレーのカラーを`Zinc900`から`Stone800`に変更してます！
カラーコードは`#292524`です

**Comment by pss-aileen** [comment]
# その他

**Reply by pss-aileen** [reply]
## テスト投稿
- 情報をまとめてみやすくするためにコメントをカテゴリーごとに投稿
- そのリプライに共有事項を足していくスタイルをとればわかりそう
- アサインがないので通知がいかない？かも？

**Comment by kagomen** [comment]
Wikiに記載したので、ピン外します！
